imports/ui/unit/unit.jsx

```diff
@@ -83,12 +114,103 @@ class Unit extends Component {
       })
     }
   }
+  handleReportFilterClicked = (event, index, reportStatus) => {
+    this.setState({
+      reportStatus: reportStatus
+    })
+  }
+
+  handleReportSortClicked = (event, index, value) => {
+    this.setState({
+      sortedReports: value
+    })
+  }
+
+  reportFilterMenu (reportStatus) {
+    const status = ['Draft', 'Finalized', 'Created By Me']
+    return status.map((name) => (
+      <MenuItem
+        key={name}
+        insetChildren
+        checked={reportStatus && reportStatus.indexOf(name) > -1}
+        value={name}
+        primaryText={name}
+      />
+    ))
+  }
+
+  reportSortMenu (sortedReports) {
+    const dateAdded = ['Newest', 'Oldest', 'Name (A to Z)', 'Name (Z to A)']
+    return dateAdded.map((elem, index) => (
+      <MenuItem
+        key={elem}
+        value={index}
+        primaryText={elem}
+      />
+    ))
+  }
+
+  filteredReportItems () {
+    const { reportList, currentUser } = this.props
+    const { reportStatus, sortedReports } = this.state
+    const statusFilter = reportStatus.length === 3 || (reportStatus.includes('Draft') && reportStatus.includes('Finalized')) ? report => true
+      : reportStatus.includes('Draft') ? report => report.status === 'UNCONFIRMED'
+        : reportStatus.includes('Finalized') ? report => report.status === 'CONFIRMED' : report => true
+    const creatorFilter = reportStatus.includes('Created By Me') ? report => report.assignee === currentUser.bugzillaCreds.login : report => true
+    const filteredReports = reportList.filter(reportItem => creatorFilter(reportItem) && statusFilter(reportItem))
+      .sort((a, b) => {
+        const dateA = new Date(a.creation_time)
```

Comment:
Use `Date.parse()` instead of the Date constructor to resolve the date string to an Epoch time integer (which you can do numeric operations on further down)